--- a/arch/mips/include/asm/pgtable.h
+++ b/arch/mips/include/asm/pgtable.h
@@ -197,7 +197,8 @@
 static inline void set_pte(pte_t *ptep, pte_t pteval)
 {
 	*ptep = pteval;
-#if !defined(CONFIG_CPU_R3K_TLB)
+#if !defined(CONFIG_CPU_R3000) && !defined(CONFIG_CPU_TX39XX) &&               \
+	!defined(CONFIG_CPU_RLX4181)
 	if (pte_val(pteval) & _PAGE_GLOBAL) {
 		pte_t *buddy = ptep_buddy(ptep);
 		/*
@@ -216,7 +217,8 @@
 static inline void pte_clear(struct mm_struct *mm, unsigned long addr, pte_t *ptep)
 {
 	htw_stop();
-#if !defined(CONFIG_CPU_R3K_TLB)
+#if !defined(CONFIG_CPU_R3000) && !defined(CONFIG_CPU_TX39XX) &&               \
+	!defined(CONFIG_CPU_RLX4181)
 	/* Preserve global status for the pair */
 	if (pte_val(*ptep_buddy(ptep)) & _PAGE_GLOBAL)
 		set_pte_at(mm, addr, ptep, __pte(_PAGE_GLOBAL));
