diff -Naur musl-1.2.5.orig/arch/mips/atomic_arch.h musl-1.2.5-patched/arch/mips/atomic_arch.h
--- musl-1.2.5.orig/arch/mips/atomic_arch.h	2025-11-23 10:31:24.692059535 +0100
+++ musl-1.2.5-patched/arch/mips/atomic_arch.h	2025-11-23 10:31:45.103429232 +0100
@@ -8,7 +8,7 @@
 static inline int a_ll(volatile int *p)
 {
 	int v;
-#if __mips < 2
+#if __mips < 2 && !(defined(__mlx4380) || defined(__mlx5380))
 	__asm__ __volatile__ (
 		".set push ; .set mips2\n\t"
 		"ll %0, %1"
@@ -26,7 +26,7 @@
 static inline int a_sc(volatile int *p, int v)
 {
 	int r;
-#if __mips < 2
+#if __mips < 2 && !(defined(__mlx4380) || defined(__mlx5380))
 	__asm__ __volatile__ (
 		".set push ; .set mips2\n\t"
 		"sc %0, %1"
@@ -43,7 +43,13 @@
 #define a_barrier a_barrier
 static inline void a_barrier()
 {
-#if __mips < 2
+#if defined(__mlx4380)
+	__asm__ __volatile__ (
+		".set push ; .set mips2 ; sync ; .set pop"
+		: : : "memory");
+#elif defined(__mlx5380)
+	__asm__ __volatile__ ("sync" : : : "memory");
+#elif __mips < 2
 	/* mips2 sync, but using too many directives causes
 	 * gcc not to inline it, so encode with .long instead. */
 	__asm__ __volatile__ (".long 0xf" : : : "memory");
diff -Naur musl-1.2.5.orig/arch/mips/crt_arch.h musl-1.2.5-patched/arch/mips/crt_arch.h
--- musl-1.2.5.orig/arch/mips/crt_arch.h	2025-11-23 10:31:24.692059535 +0100
+++ musl-1.2.5-patched/arch/mips/crt_arch.h	2025-11-23 10:31:45.103429232 +0100
@@ -16,11 +16,19 @@
 ".hidden _DYNAMIC \n"
 "	.gpword _DYNAMIC \n"
 "1:	lw $gp, 0($ra) \n"
+#if defined(__mlx4380)
+"	move $4, $sp \n"
+"	subu $gp, $ra, $gp \n"
+"	lw $5, 8($ra) \n"
+"	lw $25, 4($ra) \n"
+"	addu $5, $5, $gp \n"
+#else
 "	subu $gp, $ra, $gp \n"
 "	move $4, $sp \n"
 "	lw $5, 8($ra) \n"
 "	addu $5, $5, $gp \n"
 "	lw $25, 4($ra) \n"
+#endif
 "	addu $25, $25, $gp \n"
 "	and $sp, $sp, -8 \n"
 "	jalr $25 \n"
diff -Naur musl-1.2.5.orig/arch/mips/pthread_arch.h musl-1.2.5-patched/arch/mips/pthread_arch.h
--- musl-1.2.5.orig/arch/mips/pthread_arch.h	2025-11-23 10:31:24.692059535 +0100
+++ musl-1.2.5-patched/arch/mips/pthread_arch.h	2025-11-23 10:31:45.103429232 +0100
@@ -1,5 +1,10 @@
 static inline uintptr_t __get_tp()
 {
+#if defined(__mlx5380)
+	register uintptr_t tp;
+	__asm__ ("mflxc0 %0, $8" : "=r" (tp) );
+	return tp;
+#else
 	register uintptr_t tp __asm__("$3");
 #if __mips_isa_rev < 2
 	__asm__ (".word 0x7c03e83b" : "=r" (tp) );
@@ -7,6 +12,7 @@
 	__asm__ ("rdhwr %0, $29" : "=r" (tp) );
 #endif
 	return tp;
+#endif
 }
 
 #define TLS_ABOVE_TP
diff -Naur musl-1.2.5.orig/arch/mips/reloc.h musl-1.2.5-patched/arch/mips/reloc.h
--- musl-1.2.5.orig/arch/mips/reloc.h	2025-11-23 10:31:24.692059535 +0100
+++ musl-1.2.5-patched/arch/mips/reloc.h	2025-11-23 10:31:45.103429232 +0100
@@ -35,6 +35,7 @@
 #define CRTJMP(pc,sp) __asm__ __volatile__( \
 	"move $sp,%1 ; jr %0" : : "r"(pc), "r"(sp) : "memory" )
 
+#ifndef __mlx4380
 #define GETFUNCSYM(fp, sym, got) __asm__ ( \
 	".hidden " #sym "\n" \
 	".set push \n" \
@@ -49,3 +50,21 @@
 	"	addu %0, %0, $ra \n" \
 	".set pop \n" \
 	: "=r"(*(fp)) : : "memory", "ra" )
+#else
+#define GETFUNCSYM(fp, sym, got) __asm__ ( \
+	".hidden " #sym "\n" \
+	".set push \n" \
+	".set noreorder \n" \
+	"	bal 1f \n" \
+	"	 nop \n" \
+	"	.gpword . \n" \
+	"	.gpword " #sym " \n" \
+	"1:	lw %0, ($ra) \n" \
+	"	nop \n" \
+	"	subu %0, $ra, %0 \n" \
+	"	lw $ra, 4($ra) \n" \
+	"	nop \n" \
+	"	addu %0, %0, $ra \n" \
+	".set pop \n" \
+	: "=r"(*(fp)) : : "memory", "ra" )
+#endif 
diff -Naur musl-1.2.5.orig/arch/mips/syscall_arch.h musl-1.2.5-patched/arch/mips/syscall_arch.h
--- musl-1.2.5.orig/arch/mips/syscall_arch.h	2025-11-23 10:31:24.692059535 +0100
+++ musl-1.2.5-patched/arch/mips/syscall_arch.h	2025-11-23 10:31:45.103429232 +0100
@@ -141,11 +141,13 @@
 	return r7 && r2>0 ? -r2 : r2;
 }
 
+#ifndef __mlx4380
 #define VDSO_USEFUL
 #define VDSO_CGT32_SYM "__vdso_clock_gettime"
 #define VDSO_CGT32_VER "LINUX_2.6"
 #define VDSO_CGT_SYM "__vdso_clock_gettime64"
 #define VDSO_CGT_VER "LINUX_2.6"
+#endif
 
 #define SO_SNDTIMEO_OLD 0x1005
 #define SO_RCVTIMEO_OLD 0x1006
diff -Naur musl-1.2.5.orig/crt/mips/crti.s musl-1.2.5-patched/crt/mips/crti.s
--- musl-1.2.5.orig/crt/mips/crti.s	2025-11-23 10:31:24.700059288 +0100
+++ musl-1.2.5-patched/crt/mips/crti.s	2025-11-23 10:31:45.112428953 +0100
@@ -17,3 +17,5 @@
 	subu $sp,$sp,32
 	sw $gp,24($sp)
 	sw $ra,28($sp)
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/crt/mips/crtn.s musl-1.2.5-patched/crt/mips/crtn.s
--- musl-1.2.5.orig/crt/mips/crtn.s	2025-11-23 10:31:24.700059288 +0100
+++ musl-1.2.5-patched/crt/mips/crtn.s	2025-11-23 10:32:04.906817712 +0100
@@ -3,11 +3,25 @@
 .section .init
 	lw $gp,24($sp)
 	lw $ra,28($sp)
+#if defined(__mlx4380)
+	addu $sp,$sp,32
+	j $ra
+	nop
+#else
 	j $ra
 	addu $sp,$sp,32
+#endif
 
 .section .fini
 	lw $gp,24($sp)
 	lw $ra,28($sp)
+#if defined(__mlx4380)
+	addu $sp,$sp,32
+	j $ra
+	nop
+#else
 	j $ra
 	addu $sp,$sp,32
+#endif
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/fenv/mips/fenv.S musl-1.2.5-patched/src/fenv/mips/fenv.S
--- musl-1.2.5.orig/src/fenv/mips/fenv.S	2025-11-23 10:31:24.709059010 +0100
+++ musl-1.2.5-patched/src/fenv/mips/fenv.S	2025-11-23 10:31:45.122428644 +0100
@@ -70,3 +70,5 @@
 	li      $2, 0
 
 #endif
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/ldso/mips/dlsym.s musl-1.2.5-patched/src/ldso/mips/dlsym.s
--- musl-1.2.5.orig/src/ldso/mips/dlsym.s	2025-11-23 10:31:24.712058918 +0100
+++ musl-1.2.5-patched/src/ldso/mips/dlsym.s	2025-11-23 10:31:45.124428582 +0100
@@ -13,5 +13,10 @@
 	jalr $25
 	nop
 	lw $ra, 12($sp)
+#if defined(__mlx4380)
+	nop
+#endif
 	jr $ra
 	addiu $sp, $sp, 16
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/ldso/mips/dlsym_time64.S musl-1.2.5-patched/src/ldso/mips/dlsym_time64.S
--- musl-1.2.5.orig/src/ldso/mips/dlsym_time64.S	2025-11-23 10:31:24.712058918 +0100
+++ musl-1.2.5-patched/src/ldso/mips/dlsym_time64.S	2025-11-23 10:31:45.124428582 +0100
@@ -1,3 +1,5 @@
 #define __dlsym __dlsym_redir_time64
 #define dlsym __dlsym_time64
 #include "dlsym.s"
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/setjmp/mips/longjmp.S musl-1.2.5-patched/src/setjmp/mips/longjmp.S
--- musl-1.2.5.orig/src/setjmp/mips/longjmp.S	2025-11-23 10:31:24.733058269 +0100
+++ musl-1.2.5-patched/src/setjmp/mips/longjmp.S	2025-11-23 10:32:04.912817527 +0100
@@ -30,5 +30,13 @@
 	lw      $22, 32($4)
 	lw      $23, 36($4)
 	lw      $30, 40($4)
+#if defined(__mlx4380)
+	lw      $28, 44($4)
+	jr      $ra
+	nop
+#else
 	jr      $ra
 	lw      $28, 44($4)
+#endif
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/setjmp/mips/setjmp.S musl-1.2.5-patched/src/setjmp/mips/setjmp.S
--- musl-1.2.5.orig/src/setjmp/mips/setjmp.S	2025-11-23 10:31:24.733058269 +0100
+++ musl-1.2.5-patched/src/setjmp/mips/setjmp.S	2025-11-23 10:31:45.147427872 +0100
@@ -31,3 +31,5 @@
 #endif
 	jr      $ra
 	li      $2, 0
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/signal/mips/sigsetjmp.s musl-1.2.5-patched/src/signal/mips/sigsetjmp.s
--- musl-1.2.5.orig/src/signal/mips/sigsetjmp.s	2025-11-23 10:31:24.735058208 +0100
+++ musl-1.2.5-patched/src/signal/mips/sigsetjmp.s	2025-11-23 10:31:45.148427842 +0100
@@ -15,6 +15,9 @@
 	sw $16, 104+4+16($4)
 
 	lw $25, %call16(setjmp)($gp)
+#if defined(__mlx4380)
+	nop
+#endif
 	jalr $25
 	 move $16, $4
 
@@ -25,9 +28,17 @@
 
 .hidden __sigsetjmp_tail
 	lw $25, %call16(__sigsetjmp_tail)($gp)
+#if defined(__mlx4380)
+	nop
+#endif
 	jr $25
 	 nop
 
 1:	lw $25, %call16(setjmp)($gp)
+#if defined(__mlx4380)
+	nop
+#endif
 	jr $25
 	 nop
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/thread/mips/clone.s musl-1.2.5-patched/src/thread/mips/clone.s
--- musl-1.2.5.orig/src/thread/mips/clone.s	2025-11-23 10:31:24.743057961 +0100
+++ musl-1.2.5-patched/src/thread/mips/clone.s	2025-11-23 10:31:45.156427595 +0100
@@ -34,3 +34,5 @@
 	move $4, $2
 	li $2, 4001
 	syscall
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/thread/mips/syscall_cp.s musl-1.2.5-patched/src/thread/mips/syscall_cp.s
--- musl-1.2.5.orig/src/thread/mips/syscall_cp.s	2025-11-23 10:31:24.743057961 +0100
+++ musl-1.2.5-patched/src/thread/mips/syscall_cp.s	2025-11-23 10:31:45.156427595 +0100
@@ -17,6 +17,9 @@
 	subu    $sp, $sp, 32
 __cp_begin:
 	lw      $4, 0($4)
+#if defined(__mlx4380)
+	nop
+#endif
 	bne     $4, $0, __cp_cancel
 	move    $2, $5
 	move    $4, $6
@@ -46,8 +49,15 @@
 	.gpword .
 	.gpword __cancel
 1:	lw      $3, ($ra)
+#if defined(__mlx4380)
+	lw      $25, 4($ra)
+	subu    $3, $ra, $3
+#else
 	subu    $3, $ra, $3
 	lw      $25, 4($ra)
+#endif
 	addu    $25, $25, $3
 	jr      $25
 	move    $ra, $2
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/thread/mips/__unmapself.s musl-1.2.5-patched/src/thread/mips/__unmapself.s
--- musl-1.2.5.orig/src/thread/mips/__unmapself.s	2025-11-23 10:31:24.743057961 +0100
+++ musl-1.2.5-patched/src/thread/mips/__unmapself.s	2025-11-23 10:31:45.156427595 +0100
@@ -8,3 +8,5 @@
 	li $4, 0
 	li $2, 4001
 	syscall
+
+.section .note.GNU-stack,"",%progbits
diff -Naur musl-1.2.5.orig/src/unistd/mips/pipe.s musl-1.2.5-patched/src/unistd/mips/pipe.s
--- musl-1.2.5.orig/src/unistd/mips/pipe.s	2025-11-23 10:31:24.748057806 +0100
+++ musl-1.2.5-patched/src/unistd/mips/pipe.s	2025-11-23 10:31:45.162427409 +0100
@@ -11,6 +11,9 @@
 	beq $7, $0, 1f
 	nop
 	lw $25, %call16(__syscall_ret)($gp)
+#if defined(__mlx4380)
+	nop
+#endif
 	jr $25
 	subu $4, $0, $2
 1:	sw $2, 0($4)
@@ -18,3 +21,5 @@
 	move $2, $0
 	jr $ra
 	nop
+
+.section .note.GNU-stack,"",%progbits
